operator-pipeline-images/operatorcert/static_tests/common/catalog.py

```diff
@@ -0,0 +1,55 @@
+"""A common test suite for file based catalog"""
+
+from collections.abc import Iterator
+
+from operator_repo import OperatorCatalog, OperatorCatalogList
+from operator_repo.checks import CheckResult, Fail
+
+
+def _get_bundle_registries_from_catalog(catalog: OperatorCatalog) -> list[str]:
+    """Get bundle images from catalog"""
+    catalog_content = catalog.catalog_content
+    bundle_images = []
+    for item in catalog_content:
+        if isinstance(item, dict):
+            if item.get("schema") == "olm.bundle":
+                if image := item.get("image"):
+                    bundle_images.append(image)
+
+    return bundle_images
+
+
+def _filter_invalid_images(
+    bundle_images: list[str], allowed_registries: list[str]
+) -> list[str]:
+    """Filter out invalid images"""
+    invalid_images = []
+    for image in bundle_images:
+        if not any(image.startswith(registry) for registry in allowed_registries):
+            invalid_images.append(image)
+
+    return invalid_images
+
+
+def check_bundle_images_in_fbc(
+    operator_catalogs: OperatorCatalogList,
+) -> Iterator[CheckResult]:
+    """Validate if bundle image reference in fbc is in allowed registry"""
+
+    allowed_registries = None
+    for catalog in operator_catalogs:
+        if not isinstance(allowed_registries, list):
```

Comment:
The config is always the same for all catalogs. It doesn't make sense to me to have this in the `for` loop. It unnecessarily pollutes a code and it is hard to read. You can create a separate function to get the list of allowed catalogs and return much earlier if not present. The for loop should really focus on iterating over catalogs and checking values.

Reply:
Good point - moved outside of the `for` loop.